pxtlib/auth.ts

```diff
@@ -201,18 +229,17 @@ namespace pxt.auth {
 
             pxt.tickEvent('auth.logout');
 
-            // backend will clear the cookie token and pass back the provider logout endpoint.
-            await AuthClient.staticApiAsync('/api/auth/logout');
-
-            // Clear csrf token so we can no longer make authenticated requests.
-            await pxt.storage.shared.delAsync(AUTH_CONTAINER, CSRF_TOKEN_KEY);
+            // Tell backend to clear the http-only auth cookie. Ignore errors here, we want to clear local state even if the backend rejects this request.
+            try { await AuthClient.staticApiAsync('/api/auth/logout'); } catch {}
 
-            // Update state and UI to reflect logged out state.
-            const hash = continuationHash ? continuationHash.startsWith('#') ? continuationHash : `#${continuationHash}` : "";
+            // Clear local auth state so we can no longer make authenticated requests.
+            await delAuthTokenAsync();
+            await delUserStateAsync();
 
-            // Redirect to home screen, or skillmap home screen
+            // Redirect to home screen
             if (pxt.BrowserUtils.hasWindow()) {
-                window.location.href = `${window.location.origin}${window.location.pathname}${hash}`;
+                const hash = continuationHash ? continuationHash.startsWith('#') ? continuationHash : `#${continuationHash}` : "";
+                window.location.href = `${window.location.origin}${window.location.pathname}${window.location.search}${hash}`;
```

Comment:
What does the `location.search` do?

Reply:
It returns the query string of the current url, e.g. `?identity=1&dbg=1`.